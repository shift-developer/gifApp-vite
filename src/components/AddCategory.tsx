import { useState } from 'react';

export const AddCategory = ({ onNewCategory } : { onNewCategory: (value: string) => void }) => {

    const [ inputValue, setInputValue ] = useState<string>('');

    const onInputChange = ({ target } : React.ChangeEvent<HTMLInputElement>) => {
        setInputValue( target.value );
    }

    const onSubmit = ( event: React.FormEvent<HTMLFormElement> ) => {
        event.preventDefault();
        if( inputValue.trim().length <= 1) return;

        setInputValue('');
        onNewCategory( inputValue.trim() );
    }

    return (
        <form onSubmit={ onSubmit }>
            <input 
                type="text"
                placeholder="Find gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
    )
}