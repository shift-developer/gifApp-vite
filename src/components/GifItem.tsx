import { Gif } from '../helpers/getGifs'

export const GifItem = ({ title, url } : Gif) => {

    return (
      <div className="card">
          <img src={ url } alt={ title } />
          <p>{ title }</p>
      </div>
    )
  }