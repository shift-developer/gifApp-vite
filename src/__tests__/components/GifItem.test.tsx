
import { render, screen } from '@testing-library/react';
import { GifItem } from '../../components/GifItem';


describe('Pruebas en <GifItem />', () => {

    const title = 'Funny';
    const url   = 'https://lol.com/funny.gif';
    const id = '1';

    test('debe de hacer match con el snapshot', () => {
        
        const { container } = render( <GifItem title={ title } url={ url } id={id}/> );
        expect( container ).toMatchSnapshot();

    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        
        render( <GifItem title={ title } url={ url } id={id}/> );
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        const { src, alt} = screen.getByRole('img') as HTMLImageElement;

        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    });


    test('debe de mostrar el titulo en el componente', () => {
        
        render( <GifItem title={ title } url={ url } id={id} /> );
        expect( screen.getByText( title ) ).toBeTruthy();

    });

    
});