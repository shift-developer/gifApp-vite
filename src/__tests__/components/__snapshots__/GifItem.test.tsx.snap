// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Funny"
      src="https://lol.com/funny.gif"
    />
    <p>
      Funny
    </p>
  </div>
</div>
`;
